hydromt/metadata_resolver/convention_resolver.py

```diff
@@ -89,19 +90,24 @@ def _resolve_wildcards(
     ) -> Set[str]:
         return set(reduce(lambda uri_res, uri: uri_res + fs.glob(uri), uris, []))
 
+    @unused_kwargs_method(unused_kwargs=["mask", "zoom_level"], logger=logger)
     def resolve(
         self,
         uri: str,
         fs: AbstractFileSystem,
         *,
         time_range: Optional[TimeRange] = None,
         mask: Optional[Geom] = None,
+        zoom_level: Optional[ZoomLevel] = None,
         variables: Optional[List[str]] = None,
         handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
         logger: Logger = logger,
-        **kwargs,
     ) -> List[str]:
         """Resolve the placeholders in the URI."""
+        warn_on_unused_kwargs(
+            self, dict(zip(("mask", "zoom_level"), (mask, zoom_level))), logger
+        )
+
```

Comment:
@savente93 I used two methods of annotating unused kwargs. I can also put the `dict(zip` inside the warn function. Which one do you prefer and why?

Reply:
sorry I don't quite understand the question, can you either rephrase or call me about this? thanks

Reply:
@savente93 based on our discussion we chose to go for a simple function. I implemented this in the next commits.